pkg/shell/topnav.jsx

```diff
@@ -135,7 +132,8 @@ export class TopNav extends React.Component {
         });
 
         docItems.push(<DropdownSeparator key="separator1" />);
-        docItems.push(<DropdownItem key="about" component="button" onClick={() => this.setState({ aboutDialogOpened: true })}>
+        docItems.push(<DropdownItem key="about" component="button"
+                                    onClick={() => D.show(<AboutCockpitModal onClose={D.close} />)}>
```

Comment:
This looks interesting, thanks! What I wonder here is what `D` is. It's declared nowhere in this file. It's obviously the `const D` from inside the `const WithDialogs`, I just wonder why that doesn't have to be qualified. Does merely importing `DialogsContext` magically declare the `D` global?

Reply:
This commit is from #17298, let's discuss that first I'd say.

Reply:
The D comes from line 173, `const D = this.context`, which is the value of the closests `DialogsContext.Provider` because of line 240, `TopNavInner.contextType = DialogsContext;`.  Using React contexts with class components is quite ugly, IMO. Maybe there is a better way that I didn't see.

With function components, one writes `const D = useDialogs();` at the top of the function.